robot/impl/resource_manager.go

```diff
@@ -50,6 +51,8 @@ type resourceManager struct {
 	resources      *resource.Graph
 	processManager pexec.ProcessManager
 	processConfigs map[string]pexec.ProcessConfig
+	// modManagerLock controls access to the moduleManager
```

Comment:
[nit] Can we just elaborate a bit on comment explaining that a `Kill` from local robot may try to access the module manager concurrently with resource manager accesses?

Reply:
added